testsuite/Makefile

```diff
@@ -97,14 +99,17 @@ new:
 
 .PHONY: new-without-report
 new-without-report: lib tools
-	@for file in `$(find) tests -name ocamltests`; do \
+	@rm -f $(failstamp)
+	@(for file in `$(find) tests -name ocamltests`; do \
 	  dir=`dirname $$file`; \
 	  echo Running tests from \'$$dir\' ... ; \
-	  while read testfile; do \
+	  (while read testfile; do \
 	    TERM=dumb OCAMLRUNPARAM= \
-	      $(ocamltest) $$dir/$${testfile/$$'\r'}; \
-	  done < $$file; \
-	done 2>&1 | tee -a _log
+	      $(ocamltest) $$dir/$${testfile/$$'\r'} || \
+	      touch $(failstamp); \
+	  done < $$file) || touch $(failstamp); \
+	done || touch $(failstamp)) 2>&1 | tee -a _log
```

Comment:
There are three occurrences of `touch $(failstamp)` here. I understand the one that follows `$(ocamltest) ...` of course, but I'm not sure when the two other would fire. I was worried that they could fire if either loop did no iteration at all (but I checked bash's manual and `while/for` return 0 when no iteration happens), they return the return code of the last command of the last iteration. So I guess they are useless for now, but they can be seen as defensive programming (if we add extra commands after the ocamltest invocation, we will know if they fail).